x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/jinaai/JinaAIResponseHandler.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.jinaai;
+
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.inference.InferenceServiceResults;
+import org.elasticsearch.xpack.core.inference.results.StreamingChatCompletionResults;
+import org.elasticsearch.xpack.inference.external.http.HttpResult;
+import org.elasticsearch.xpack.inference.external.http.retry.BaseResponseHandler;
+import org.elasticsearch.xpack.inference.external.http.retry.ResponseParser;
+import org.elasticsearch.xpack.inference.external.http.retry.RetryException;
+import org.elasticsearch.xpack.inference.external.request.Request;
+import org.elasticsearch.xpack.inference.external.response.jinaai.JinaAIErrorResponseEntity;
+import org.elasticsearch.xpack.inference.logging.ThrottlerManager;
+
+import java.util.concurrent.Flow;
+
+import static org.elasticsearch.xpack.inference.external.http.HttpUtils.checkForEmptyBody;
+
+/**
+ * Defines how to handle various errors returned from the JinaAI integration.
+ *
+ */
+public class JinaAIResponseHandler extends BaseResponseHandler {
+    static final String INPUTS_ARRAY_TOO_LARGE_MESSAGE_MATCHER = "is larger than the largest allowed size";
+    static final String INPUTS_ARRAY_ERROR_MESSAGE = "Received an inputs array too large response";
+    static final String PAYMENT_ERROR_MESSAGE = "Payment required";
+
+    public JinaAIResponseHandler(String requestType, ResponseParser parseFunction) {
+        super(requestType, parseFunction, JinaAIErrorResponseEntity::fromResponse);
+    }
+
+    @Override
+    public void validateResponse(ThrottlerManager throttlerManager, Logger logger, Request request, HttpResult result)
+        throws RetryException {
+        checkForFailureStatusCode(request, result);
+        checkForEmptyBody(throttlerManager, logger, request, result);
+    }
+
+    @Override
+    public boolean canHandleStreamingResponses() {
+        return false;
+    }
+
+    @Override
+    public InferenceServiceResults parseResult(Request request, Flow.Publisher<HttpResult> flow) {
+        // TODO (JoanFM): Check this logic
+        return new StreamingChatCompletionResults(null);
+    }
```

Comment:
```suggestion
```

I think you can remove these two overrides.

The top interface [ResponseHandler](https://github.com/elastic/elasticsearch/blob/d09d57db802ecba45195abfd1b38c12818bbe537/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/retry/ResponseHandler.java#L45-L74) in the inheritance chain provides default implementations for both `canHandleStreamingResponses` and `parseResult` handling providers, which do not handle streaming responses.